adam-core/src/main/scala/org/bdgenomics/adam/rdd/variant/VariantContextRDD.scala

```diff
@@ -227,8 +237,20 @@ case class VariantContextRDD(rdd: RDD[VariantContext],
                 deferMerging: Boolean,
                 disableFastConcat: Boolean,
                 stringency: ValidationStringency) {
-    val vcfFormat = VCFFormat.inferFromFilePath(filePath)
-    assert(vcfFormat == VCFFormat.VCF, "BCF not yet supported") // TODO: Add BCF support
+
+    // TODO: Add BCF support
+    val vcfFormat = VCFFormat.VCF
+    val isBgzip = FileExtensions.isGzip(filePath)
+    if (!FileExtensions.isVcfExt(filePath)) {
+      val msg = "Saw non-VCF extension for file %s. Extensions supported are .vcf<.gz, .bgz>"
+        .format(filePath)
+
+      if (stringency == ValidationStringency.STRICT) {
+        throw new IllegalArgumentException(msg)
+      } else {
+        log.warn(msg)
```

Comment:
Does it make sense to stringency this?  If we warn here, does the file still get written correctly?